### ISSUE TITLE
[bug] Safe error ether v5

### ISSUE THREAD

**Issue by CjoyHome** [WalletConnect/web3modal#1574]
### Link to minimal reproducible example

https://github.com/WalletConnect/web3modal/issues/1552

### Summary

Hello, the prompt box on the safe side no longer pops up, but now there is a new problem. When testing https://lab.web3modal.com/library/ethers-siwe/, the error is reported as follows
My project error:
<img width="1072" alt="image" src="https://github.com/WalletConnect/web3modal/assets/26813427/cd382a77-c01a-429f-b5ce-9cda52486c0e">
<img width="1165" alt="image" src="https://github.com/WalletConnect/web3modal/assets/26813427/5c1f935b-ee87-4259-94f2-822c9dd28b38">

Ether with SIWE demo error:
<img width="1035" alt="image" src="https://github.com/WalletConnect/web3modal/assets/26813427/99d2c25a-fd8b-47f1-b051-fe6a758fac28">

### List of related npm package versions

version：

```
  "@walletconnect/modal": "^2.6.2",
    "@web3modal/ethers5": "^3.4.0",
    "ethers": "^5.7.2"
```

**Comment by CjoyHome** [comment]
@glitch-txs can you help me? 😭

**Comment by glitch-txs** [comment]
It worked for me, could you try clearing cache?

**Comment by CjoyHome** [comment]
I cleared the cache but still the same problem

**Comment by glitch-txs** [comment]
can you upgrade to 3.5?

**Comment by CjoyHome** [comment]
When I updated @web3modal/ethers5 to 3.5, an error occurred
![image](https://github.com/WalletConnect/web3modal/assets/26813427/9917df0a-6405-486d-958d-5c9e5d710017)

**Comment by glitch-txs** [comment]
it's working fine on my end, could you set up a minimal reproducible example and open a new issue for this one please

**Comment by CjoyHome** [comment]
OK, I have opened a new issue

**Comment by glitch-txs** [comment]
@CjoyHome this issue is from Safe, there's a problem with signatures, please open an issue in their repository, thanks!

**Comment by CjoyHome** [comment]
> ### Link to minimal reproducible example
> 链接到最小可重现示例
> #1552
> 
> ### Summary 概括
> Hello, the prompt box on the safe side no longer pops up, but now there is a new problem. When testing https://lab.web3modal.com/library/ethers-siwe/, the error is reported as follows您好，安全端的提示框不再弹出了，但是现在又出现了新的问题。测试https://lab.web3modal.com/library/ethers-siwe/时，报错如下 My project error: 我的项目错误： <img alt="image" width="1072" src="https://private-user-images.githubusercontent.com/26813427/290722636-cd382a77-c01a-429f-b5ce-9cda52486c0e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5OTY4MDQsIm5iZiI6MTcwMjk5NjUwNCwicGF0aCI6Ii8yNjgxMzQyNy8yOTA3MjI2MzYtY2QzODJhNzctYzAxYS00MjlmLWI1Y2UtOWNkYTUyNDg2YzBlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE5VDE0MzUwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjMDNjZWJkNDJlYmVmMGM0MTE3NmM3YWE1ZWUxMGY4MDIwODdhMzI1ZTBiMjhjYTFmNmY5MGE3OWVjZjQ4M2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.863i03aJNBFRMDYREfYTM9N9vm10OAmf8eDiRwQOBB8"> <img alt="image" width="1165" src="https://private-user-images.githubusercontent.com/26813427/290722646-5c1f935b-ee87-4259-94f2-822c9dd28b38.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5OTY4MDQsIm5iZiI6MTcwMjk5NjUwNCwicGF0aCI6Ii8yNjgxMzQyNy8yOTA3MjI2NDYtNWMxZjkzNWItZWU4Ny00MjU5LTk0ZjItODIyYzlkZDI4YjM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE5VDE0MzUwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU2MzZiMmYxODI3NDQ4ODFkMjliOGNiZTY5NDI5ODA5OTFlZDFjMzk4MGI0OGViY2I3NDQ4YTZhMmU1NTAwMzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.CPmH0P43JMU4Xticea0xINiPop5otTvgUNzDjm6F1fQ">
> 
> Ether with SIWE demo error:以太坊 SIWE 演示错误： <img alt="image" width="1035" src="https://private-user-images.githubusercontent.com/26813427/290722845-99d2c25a-fd8b-47f1-b051-fe6a758fac28.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI5OTY4MDQsIm5iZiI6MTcwMjk5NjUwNCwicGF0aCI6Ii8yNjgxMzQyNy8yOTA3MjI4NDUtOTlkMmMyNWEtZmQ4Yi00N2YxLWIwNTEtZmU2YTc1OGZhYzI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE5VDE0MzUwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3ODUyZjI2ODdjZWY1Y2UxNzY0OWU1MjhlZjdjODdjYjMzZTVkMDYzZTVkZWIwZGQ2NTA2YmEzOGMwYmRhODAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9vdiHRpMnBMqQgaHpOy0GIT9jT8zqTqcPGhwawcvY9Q">
> 
> ### List of related npm package versions
> 相关 npm 包版本列表
> version： 版本：
> 
> ```
>   "@walletconnect/modal": "^2.6.2",
>     "@web3modal/ethers5": "^3.4.0",
>     "ethers": "^5.7.2"
> ```


I reported this issue to Safe and their response was as follows
![image](https://github.com/WalletConnect/web3modal/assets/26813427/21878fe9-848e-436b-8cfd-cdd1c6959517)

Can this problem be solved by upgrading to 3.5.0? Do I need to wait for #1584 to be solved now?

**Comment by glitch-txs** [comment]
could you share the link of the Safe issue please?

**Comment by glitch-txs** [comment]
We have this fix for 3.6 https://github.com/WalletConnect/web3modal/pull/1553
But Safe should still work since it's optional chains param

**Comment by CjoyHome** [comment]
Because I don’t know how to reply, the other party has closed
https://github.com/safe-global/safe-wallet-web/issues/3037
